pkg/ccl/changefeedccl/checkpoint/metrics.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package checkpoint
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/util/metric"
+	"github.com/cockroachdb/cockroach/pkg/util/metric/aggmetric"
+)
+
+var (
+	metaCreateNanos = metric.Metadata{
+		Name:        "changefeed.checkpoint.create_nanos",
+		Help:        "Time it takes to create a changefeed checkpoint",
+		Unit:        metric.Unit_NANOSECONDS,
+		Measurement: "Nanoseconds",
+	}
+
+	metaTotalBytes = metric.Metadata{
+		Name:        "changefeed.checkpoint.total_bytes",
+		Help:        "Total size of a changefeed checkpoint",
+		Unit:        metric.Unit_BYTES,
+		Measurement: "Bytes",
+	}
+
+	metaSpanCount = metric.Metadata{
+		Name:        "changefeed.checkpoint.span_count",
+		Help:        "Number of spans in a changefeed checkpoint",
+		Unit:        metric.Unit_COUNT,
+		Measurement: "Spans",
+	}
+)
+
+type AggMetrics struct {
+	CreateNanos *aggmetric.AggHistogram
+	TotalBytes  *aggmetric.AggHistogram
+	SpanCount   *aggmetric.AggHistogram
```

Comment:
I question whether these two have to be histograms, for cardinality / metric count reasons. But if you think it's worth it then it's ok with me.

Reply:
We discussed this offline but my thinking is that it would be useful as a histogram so we can see how the average checkpoint size has changed over time.